jvm/src/main/scala/kiama/util/Server.scala

```diff
@@ -533,6 +549,31 @@ trait LanguageService[N] {
   def getReferences(position: Position, includeDecl: Boolean): Option[Vector[N]] =
     None
 
+  /**
+   * Return the inlay hints in the range (if any).
+   * Default is to never return anything.
+   *
+   * TODO: Remove debug info :)
+   */
+  def getInlayHints(range: Range): Option[Vector[InlayHint]] =
+    println(s"Getting inlay hints for $range")
+    Option(Vector(
+      // Type hint example - shows type after a variable declaration
+      InlayHint(
+        kind = InlayHintKind.Type,
+        position = Position(1, 1, range.from.source), // after variable name
+        label = ": Int",
+        Some("The inferred type of this variable is Int") // tooltip on hover
+      ),
+
+      // Parameter hint example - shows parameter names at call sites
+      InlayHint(
+        kind = InlayHintKind.Parameter,
+        position = Position(2, 1, range.from.source), // before argument
+        label = "count=",
+        Some("Parameter 'count' of type Int") // tooltip on hover
+      )))
```

Comment:
Temporary, we should let Effekt override this instead, I was just too tired and lazy :)

Reply:
<img width="223" alt="Screenshot 2025-02-15 at 10 15 15" src="https://github.com/user-attachments/assets/7ca098f9-3bbc-4561-8146-9f26b06a94ff" />